lib/Dialect/BGV/CMakeLists.txt

```diff
@@ -0,0 +1,22 @@
+add_subdirectory(IR)
+add_subdirectory(Transforms)
+add_mlir_dialect_library(MLIRBGV
+        IR/BGVDialect.cpp
+        Transforms/AddClientInterface.cpp
```

Comment:
I'm not sure if this is an issue, but in Bazel, the transforms are in their own target (and therefore library), whereas in the current cmake, they're being combined into a single target.
One downside to this that I could forsee is keeping dependency lists in sync between Bazel and CMake, as CMake would have some of the dependencies "hidden".

Reply:
Oh, and I'm also wondering if there's anyway we can use `add_mlir_dialect` for the dialects themseves, as thathandles a LOT of the tablegen/etc stuff internally. See https://mlir.llvm.org/docs/Tutorials/CreatingADialect/#tablegen-targets and https://github.com/llvm/llvm-project/blob/main/mlir/cmake/modules/AddMLIR.cmake#L162 for what it does. I _think_ it continues gracefully if any of the files are missing (at least, I used it in HECO for dialects that only had a single *.td).

Alternatively, maybe we can define our own add_heir_dialect that works with our, slightly different, directory structure. My hope is that the simpler we make each CMakeLists.txt file, the more likely we'll be able to keep everything nicely in sync between Bazel and CMake

Reply:
we had this conversation on Discord; you mentioned you didn't want to touch source files nor the BUILD Bazel files. Indeed my original implementation used the add_mlir_dialect macros in CMake but I had to remove them in place of piecemeal sub-macros to enable the builds to match the unique uses of the HEIR project.

As for the  sub targets in MLIRBGV I should go back and do what is done in BUILD folder; this horrible hack is a side-effect of rushing to get this to work

Reply:
Let's merge this PR as-is, and I'll create an issue for me to wirte a custom set of `add_heir_{dialect/conversion/transform/analysis}` CMake macros. With those in place (and updating the right variables), we can then also simplify the tools build to be as simple as the upstream ones.

Reply:
sounds good to me - thanks @AlexanderViand-Intel